### ISSUE TITLE
Support video volumes

### ISSUE THREAD

**Issue by mzur** [biigle/largo#69]
Largo could be implemented for video volumes, too. Instead of a static annotation patch, the patches could be short video clips that show the annotated object (if the annotation has a duration). For single frame annotations, the patches are still images. The short video clips play on mouseover. The remaining behavior of Largo would remain unchanged. Project Largo would need to support both image annotations and video annotations.

**Comment by mzur** [comment]
Here is one way to dynamically crop a video: https://video.stackexchange.com/a/29182

**Comment by mzur** [comment]
There are two ways the video annotation thumbnails could be implemented:

1. Initially show a static thumbnail. On mouseover, wait *x* seconds, then begin to load the video clip of the annotation. Once the clip is loaded, play the clip **OR** jump to the time of the clip as indicated by the mouse position relative to the video element (similar to existing volume/video preview thumbnails). 

2. Initially show a static thumbnail. On mouseover, show a sequence (e.g. 10) of predetermined static thumbnails (similar to volume/video preview thumbnails) that show the video annotation changing over time. Jump between thumbnails as indicated by the mouse position relative to the annotation thumbnail element.

**Comment by mzur** [comment]
We decided to use option 2. This should also result in much reduced storage space requirements. Also, the UX might be better compared to a full video clip.

**Comment by mzur** [comment]
I chose to display only the first frame of a video annotation in Largo. This is more consistent with image annotations and single frame video annotations and might be less confusing when the user moves the mouse over the patches. As I already started to implement animated patches, I moved the code to the [`animated-video-patches`](https://github.com/biigle/largo/tree/animated-video-patches) branch in case it is needed sometime in the future.
